src/modules/rover_pos_control/RoverPositionControl.cpp

```diff
@@ -347,40 +378,50 @@ RoverPositionControl::run()
 
 			matrix::Vector3f ground_speed(_global_pos.vel_n, _global_pos.vel_e,  _global_pos.vel_d);
 			matrix::Vector2f current_position((float)_global_pos.lat, (float)_global_pos.lon);
+			matrix::Vector2f current_velocity(_local_pos.vx, _local_pos.vy);
 
 			// This if statement depends upon short-circuiting: If !manual_mode, then control_position(...)
 			// should not be called.
 			// It doesn't really matter if it is called, it will just be bad for performance.
-			if (!manual_mode && control_position(current_position, ground_speed, _pos_sp_triplet)) {
 
-				/* XXX check if radius makes sense here */
-				float turn_distance = _param_l1_distance.get(); //_gnd_control.switch_distance(100.0f);
+			if (!manual_mode && _control_mode.flag_control_position_enabled) {
 
-				// publish status
-				position_controller_status_s pos_ctrl_status = {};
+				if (control_position(current_position, ground_speed, _pos_sp_triplet)) {
```

Comment:
There is no `else` for the case where `control_position` fails. Does that make sense?

Reply:
I guess, the only difference is that it doesn't publish the position control status. If we want something else, shouldn't it be a separate PR?

Reply:
ok fine.